gradio/components/login_button.py

```diff
@@ -96,10 +96,11 @@ def _check_login_status(self, request: Request) -> LoginButton:
 _js_open_if_not_logged_in = """
 (buttonValue) => {
     if (!buttonValue.includes("Signed in")) {
+        url = '/login/huggingface' + window.location.search
         if ( window !== window.parent ) {
-            window.open('/login/huggingface', '_blank');
+            window.open(url, '_blank');
         } else {
-            window.location.assign('/login/huggingface');
+            window.location.assign(url);
         }
     }
```

Comment:
Once we merge https://github.com/gradio-app/gradio/pull/7063, we should make a similar change to `/logout` as well to preserve the query parameters when a user logs out

Reply:
#7063 is merged => updated in https://github.com/gradio-app/gradio/pull/7034/commits/3c11ea4b0b8cb840dedcb397a1931056e1c27ada